### PR TITLE
Aladin image zooms when changing constraints

### PR DESCRIPTION
**Pull request by cquiroz**
There is a cycle with Aladin where you ask it to use a fov (x) and it sends you back what fov is actually using (y). They don't always match but we need to get the real value to react to zoom changes. The small differences add up to produce undesired zoom when e.g. resizing. Surprisingly this is more evident for Linux users.

This PR essentially skip recording very small fov changes.
Also it was noticed that the aladin window would flash in certain cases because of unnecessary re renderings.
To solve these `Reusability` was reintroduced.
I tested this quite a bit but there maybe some regression in certain cases

### CODE REVIEW

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/targeteditor/AladinContainer.scala [hunk 1]
```diff
@@ -239,16 +244,7 @@ object AladinContainer {
       }
       // Use fov from aladin
       .useState(none[Fov])
-      // full screen trigger reflow
-      .useEffectWithDepsBy((props, _, _, _, _, _, _, _) => props.options.fullScreen)(
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targeteditor/AladinContainer.scala:243]: Why is this no longer needed?

Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/AladinContainer.scala:243]: Full screen triggers a redraw when the parent changes it. This only made the flashing effect worse by doing 2 extra repaints

**Comment by rpiaggio** [comment]
This looks like a good use for reusability :+1:

**Pull request merged**